[2025-08-02T03:47:57.105+0000] {logging_mixin.py:188} INFO - Changing /opt/***/logs/dag_id=cdr_pipeline/run_id=manual__2025-08-02T03:41:43.683798+00:00/task_id=load_to_redshift permission to 509
[2025-08-02T03:47:57.163+0000] {logging_mixin.py:188} INFO - Changing /opt/***/logs/dag_id=cdr_pipeline/run_id=manual__2025-08-02T03:41:43.683798+00:00/task_id=load_to_redshift permission to 509
[2025-08-02T03:47:57.227+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: cdr_pipeline.load_to_redshift manual__2025-08-02T03:41:43.683798+00:00 [queued]>
[2025-08-02T03:47:57.246+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: cdr_pipeline.load_to_redshift manual__2025-08-02T03:41:43.683798+00:00 [queued]>
[2025-08-02T03:47:57.247+0000] {taskinstance.py:2170} INFO - Starting attempt 3 of 4
[2025-08-02T03:47:57.273+0000] {taskinstance.py:2191} INFO - Executing <Task(S3ToRedshiftOperator): load_to_redshift> on 2025-08-02 03:41:43.683798+00:00
[2025-08-02T03:47:57.293+0000] {standard_task_runner.py:60} INFO - Started process 3220 to run task
[2025-08-02T03:47:57.301+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'cdr_pipeline', 'load_to_redshift', 'manual__2025-08-02T03:41:43.683798+00:00', '--job-id', '136', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmpxiqpg3j8']
[2025-08-02T03:47:57.303+0000] {standard_task_runner.py:88} INFO - Job 136: Subtask load_to_redshift
[2025-08-02T03:47:57.401+0000] {logging_mixin.py:188} INFO - Changing /opt/***/logs/dag_id=cdr_pipeline/run_id=manual__2025-08-02T03:41:43.683798+00:00/task_id=load_to_redshift permission to 509
[2025-08-02T03:47:57.424+0000] {task_command.py:423} INFO - Running <TaskInstance: cdr_pipeline.load_to_redshift manual__2025-08-02T03:41:43.683798+00:00 [running]> on host f2527419f673
[2025-08-02T03:47:57.574+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='cdr_pipeline' AIRFLOW_CTX_TASK_ID='load_to_redshift' AIRFLOW_CTX_EXECUTION_DATE='2025-08-02T03:41:43.683798+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-02T03:41:43.683798+00:00'
[2025-08-02T03:47:57.591+0000] {base.py:83} INFO - Using connection ID 'aws_default' for task execution.
[2025-08-02T03:47:57.602+0000] {base.py:83} INFO - Using connection ID 'aws_default' for task execution.
[2025-08-02T03:47:57.605+0000] {connection_wrapper.py:378} INFO - AWS Connection (conn_id='aws_default', conn_type='aws') credentials retrieved from login and password.
[2025-08-02T03:47:57.631+0000] {s3_to_redshift.py:192} INFO - Executing COPY command...
[2025-08-02T03:47:57.644+0000] {base.py:83} INFO - Using connection ID 'redshift_default' for task execution.
[2025-08-02T03:47:57.646+0000] {connection_wrapper.py:378} INFO - AWS Connection (conn_id='redshift_default', conn_type='aws') credentials retrieved from login and password.
[2025-08-02T03:47:57.647+0000] {warnings.py:109} WARNING - <string>:9: AirflowProviderDeprecationWarning: Host tf-redshift-cluster.ci896s3qzt6h.eu-west-1.redshift.amazonaws.com:5439/dev_db specified in the connection is not used. Please, set it on extra['endpoint_url'] instead

[2025-08-02T03:47:59.612+0000] {redshift_data.py:113} INFO - Polling statement beb902ec-c188-4c39-b801-d5058fe09b0b
[2025-08-02T03:47:59.804+0000] {redshift_data.py:129} INFO - Query STARTED
[2025-08-02T03:47:10.832+0000] {redshift_data.py:113} INFO - Polling statement beb902ec-c188-4c39-b801-d5058fe09b0b
[2025-08-02T03:48:12.855+0000] {redshift_data.py:129} INFO - Query STARTED
[2025-08-02T03:48:22.865+0000] {redshift_data.py:113} INFO - Polling statement beb902ec-c188-4c39-b801-d5058fe09b0b
[2025-08-02T03:48:23.227+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/transfers/s3_to_redshift.py", line 194, in execute
    redshift_hook.execute_query(sql=sql, **self.redshift_data_api_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/hooks/redshift_data.py", line 107, in execute_query
    self.wait_for_results(statement_id, poll_interval=poll_interval)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/hooks/redshift_data.py", line 124, in wait_for_results
    raise ValueError(
ValueError: Statement 'beb902ec-c188-4c39-b801-d5058fe09b0b' terminated with status FAILED. Response details: {'ClusterIdentifier': 'tf-redshift-cluster',
 'CreatedAt': datetime.datetime(2025, 8, 2, 3, 48, 1, 118000, tzinfo=tzlocal()),
 'Database': 'dev_db',
 'DbUser': 'masterdb',
 'Duration': -1,
 'Error': 'ERROR: Spectrum Scan Error\n'
          '  Detail: \n'
          '  -----------------------------------------------\n'
          '  error:  Spectrum Scan Error\n'
          '  code:      15007\n'
          '  context:   File '
          "'https://s3.eu-west-1.amazonaws.com/cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/66046c9b895e438284f3fd0425f0fef6.snappy.parquet' "
          'has an incompatible Parquet schema for column '
          "'s3://cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/.call_start'. "
          'Column\n'
          '  query:     209449\n'
          '  location:  dory_util.cpp:1776\n'
          '  process:   worker_thread [pid=7074]\n'
          '  -----------------------------------------------\n',
 'HasResultSet': False,
 'Id': 'beb902ec-c188-4c39-b801-d5058fe09b0b',
 'QueryString': '\n'
                '                    COPY public.cdr_records \n'
                '                    FROM '
                "'s3://cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/'\n"
                '                    credentials\n'
                "                    ''\n"
                '                    \n'
                '                    FORMAT AS PARQUET;\n'
                '        ',
 'RedshiftPid': 1073889417,
 'RedshiftQueryId': 209450,
 'ResponseMetadata': {'HTTPHeaders': {'content-length': '1183',
                                      'content-type': 'application/x-amz-json-1.1',
                                      'date': 'Sat, 02 Aug 2025 03:48:23 GMT',
                                      'x-amzn-requestid': '199b42e9-82d2-4396-a134-13a1941aff9d'},
                      'HTTPStatusCode': 200,
                      'RequestId': '199b42e9-82d2-4396-a134-13a1941aff9d',
                      'RetryAttempts': 0},
 'ResultRows': -1,
 'ResultSize': -1,
 'Status': 'FAILED',
 'UpdatedAt': datetime.datetime(2025, 8, 2, 3, 48, 17, 738000, tzinfo=tzlocal())}
[2025-08-02T03:48:23.252+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=cdr_pipeline, task_id=load_to_redshift, execution_date=20250802T034143, start_date=20250802T034757, end_date=20250802T034823
[2025-08-02T03:48:23.277+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 136 for task load_to_redshift (Statement 'beb902ec-c188-4c39-b801-d5058fe09b0b' terminated with status FAILED. Response details: {'ClusterIdentifier': 'tf-redshift-cluster',
 'CreatedAt': datetime.datetime(2025, 8, 2, 3, 48, 1, 118000, tzinfo=tzlocal()),
 'Database': 'dev_db',
 'DbUser': 'masterdb',
 'Duration': -1,
 'Error': 'ERROR: Spectrum Scan Error\n'
          '  Detail: \n'
          '  -----------------------------------------------\n'
          '  error:  Spectrum Scan Error\n'
          '  code:      15007\n'
          '  context:   File '
          "'https://s3.eu-west-1.amazonaws.com/cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/66046c9b895e438284f3fd0425f0fef6.snappy.parquet' "
          'has an incompatible Parquet schema for column '
          "'s3://cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/.call_start'. "
          'Column\n'
          '  query:     209449\n'
          '  location:  dory_util.cpp:1776\n'
          '  process:   worker_thread [pid=7074]\n'
          '  -----------------------------------------------\n',
 'HasResultSet': False,
 'Id': 'beb902ec-c188-4c39-b801-d5058fe09b0b',
 'QueryString': '\n'
                '                    COPY public.cdr_records \n'
                '                    FROM '
                "'s3://cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/'\n"
                '                    credentials\n'
                "                    ''\n"
                '                    \n'
                '                    FORMAT AS PARQUET;\n'
                '        ',
 'RedshiftPid': 1073889417,
 'RedshiftQueryId': 209450,
 'ResponseMetadata': {'HTTPHeaders': {'content-length': '1183',
                                      'content-type': 'application/x-amz-json-1.1',
                                      'date': 'Sat, 02 Aug 2025 03:48:23 GMT',
                                      'x-amzn-requestid': '199b42e9-82d2-4396-a134-13a1941aff9d'},
                      'HTTPStatusCode': 200,
                      'RequestId': '199b42e9-82d2-4396-a134-13a1941aff9d',
                      'RetryAttempts': 0},
 'ResultRows': -1,
 'ResultSize': -1,
 'Status': 'FAILED',
 'UpdatedAt': datetime.datetime(2025, 8, 2, 3, 48, 17, 738000, tzinfo=tzlocal())}; 3220)
[2025-08-02T03:48:23.357+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-08-02T03:48:23.399+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
